toolbox/oncall_alert/main.go

```diff
@@ -172,91 +175,108 @@ func formatMessage(failures map[*postSubmitJob]bool) (mess string) {
 func sendMessage(body string) {
 	msg := fmt.Sprintf("From: %s\n", sender) +
 		fmt.Sprintf("To: %s\n", receivers) +
-		fmt.Sprintf("Subject: %s [%s]\n\n", messageSubject, time.Now().In(location).Format("2006-01-02 15:04:05 PST")) +
+		fmt.Sprintf("Subject: %s [%s]\n\n", messageSubject,
+			time.Now().In(location).Format("2006-01-02 15:04:05 PST")) +
 		messagePrologue + body + messageEnding
 
 	gmailSMTPAddr := fmt.Sprintf("%s:%d", gmailSMTPSERVER, gmailSMTPPORT)
 	err := smtp.SendMail(gmailSMTPAddr, smtp.PlainAuth("istio-bot", sender, gmailAppPass, gmailSMTPSERVER),
 		sender, receivers, []byte(msg))
-
 	if err != nil {
-		log.Printf("smtp error: %s", err)
+		log.Printf("smtp error: %s\n", err)
 		return
 	}
-
-	log.Print("Alert message sent!")
+	log.Printf("Alert message sent!\n")
 }
 
 // Read latest run from "latest-build.txt" of a job under gcs "istio-prow" bucket
 // Example: istio-prow/istio-postsubmit/latest-build.txt
-func getLatestRun(job string) (int, error) {
-	lastBuildFile := filepath.Join(job, lastBuildTXT)
-	latestBuildString, err := gcsClient.GetFileFromGCSString(*gcsBucket, lastBuildFile)
+func getLatestRun(jobName string) (int, error) {
+	lastBuildFile := filepath.Join(jobName, lastBuildTXT)
+	latestBuildString, err := gcsClient.Read(*gcsBucket, lastBuildFile)
 	if err != nil {
 		return 0, err
 	}
 	latestBuildInt, err := strconv.Atoi(latestBuildString)
 	if err != nil {
-		log.Printf("Failed to convert %s to int: %v", latestBuildString, err)
+		log.Printf("Failed to convert %s to int: %v\n", latestBuildString, err)
 		return 0, err
 	}
 	return latestBuildInt, nil
 }
 
-// Check the latest run of each protected Post-Submit tests.
+// From last check onward, check all the runs of protected Post-Submit tests.
 // Record failure if it hasn't been recorded and update latestRun if necessary
-func updatePostSubmitStatus() (map[*postSubmitJob]bool, int) {
-	// If a "new" failure appears in this circle, put it in and with value true
-	// So if no new failure, this map should be empty --> no new notification
-	newFailures := make(map[*postSubmitJob]bool)
-
-	// Total count of jobs which is in failing status right now.
-	// Including jobs failed in circles before.
-	// If 0 --> unblock auto-merge
-	failedCount := 0
-
-	for _, job := range postSubmitJobs {
-		latestRunNo, err := getLatestRun(job.name)
+// Returns an array of postSubmitJob since the same job might fail at multiple runs
+func checkOnJobsWatched() []failure {
+	newFailures := []failure{}
```

Comment:
prefer var newFailures []failure. That way user can check if the value return by the function is nil as opposed to being empty.

Reply:
Done